code/hsec-tools/src/Security/OSV.hs

```diff
@@ -0,0 +1,223 @@
+-- | This module contains the OSV datatype and its ToJSON instance.
+-- The module was initialized with http://json-to-haskell.chrispenner.ca/
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE OverloadedStrings #-}
+module Security.OSV where
+
+import Data.Aeson (ToJSON(..), FromJSON(..), Value(..), (.:), (.=), object)
+import Data.Aeson.Types (prependFailure, typeMismatch)
+import Data.Text (Text)
+
+data Affected = Affected
+  { affectedRanges :: [Ranges]
+  , affectedPackage :: Package
+  , affectedEcosystemSpecific :: EcosystemSpecific
+  , affectedDatabaseSpecific :: DatabaseSpecific
+  } deriving (Show, Eq, Ord)
+
+data Affects = Affects
+  { affectsOs :: [Value]
+  , affectsArch :: [Value]
+  , affectsFunctions :: [Value]
+  } deriving (Show, Eq, Ord)
+
+data DatabaseSpecific = DatabaseSpecific
+  { databaseSpecificCvss :: Maybe Value
+  , databaseSpecificCategories :: [Value]
+  , databaseSpecificInformational :: Text
+  } deriving (Show, Eq, Ord)
+
+data EcosystemSpecific = EcosystemSpecific
+  { ecosystemSpecificAffects :: Affects
+  } deriving (Show, Eq, Ord)
+
+data Events = Events
+  { eventsIntroduced :: Text
+  } deriving (Show, Eq, Ord)
+
+data Model = Model
+  { modelDetails :: Text
+  , modelId :: Text
+  , modelSummary :: Text
+  , modelRelated :: [Value]
+  , modelAffected :: [Affected]
+  , modelAliases :: [Value]
+  , modelPublished :: Text
+  , modelReferences :: [References]
+  , modelSeverity :: [Value]
```

Comment:
we should expand the severity field to a record with `type` and `score`, or even a special purpose datatype for the registered types (`CVSS_V2` and `CVSS_V3`)

Reply:
+1